Content.Server/GameObjects/Components/Kitchen/KitchenMicrowaveComponent.cs

```diff
@@ -0,0 +1,389 @@
+            {
+                _solidsVisualList.Add(item.Uid);
+            }
+
+            _userInterface.SetState(new MicrowaveUpdateUserInterfaceState(_solution.Solution.Contents.ToList(), _solidsVisualList));
+        }
+
+        void IActivate.Activate(ActivateEventArgs eventArgs)
+        {
+            if (!eventArgs.User.TryGetComponent(out IActorComponent actor) || !Powered)
+            {
+                return;
+            }
+            UpdateUserInterface();
+            _userInterface.Open(actor.playerSession);
+
+        }
+
+        public bool AttackBy(AttackByEventArgs eventArgs)
+        {
+            var itemEntity = eventArgs.User.GetComponent<HandsComponent>().GetActiveHand.Owner;
+
+            if(itemEntity.TryGetComponent<PourableComponent>(out var attackPourable))
+            {
+                //Get target and check if it can be poured into
+                if (!Owner.TryGetComponent<SolutionComponent>(out var mySolution)
+                    || !mySolution.CanPourIn)
+                {
+                    return false;
+                }
+
+                if (!itemEntity.TryGetComponent<SolutionComponent>(out var attackSolution)
+                    || !attackSolution.CanPourOut)
+                {
+                    return false;
+                }
+
+                //Get transfer amount. May be smaller than _transferAmount if not enough room
+                var realTransferAmount = ReagentUnit.Min(attackPourable.TransferAmount, mySolution.EmptyVolume);
+                if (realTransferAmount <= 0) //Special message if container is full
+                {
+                    _notifyManager.PopupMessage(Owner.Transform.GridPosition, eventArgs.User,
+                        Loc.GetString("Container is full"));
+                    return false;
+                }
+
+                //Move units from attackSolution to targetSolution
+                var removedSolution = attackSolution.SplitSolution(realTransferAmount);
+                if (!mySolution.TryAddSolution(removedSolution))
+                {
+                    return false;
+                }
+
+                _notifyManager.PopupMessage(Owner.Transform.GridPosition, eventArgs.User,
+                    Loc.GetString("Transferred {0}u", removedSolution.TotalVolume));
+                return true;
+            }
+
+            if (!itemEntity.TryGetComponent(typeof(FoodComponent), out var food))
+            {
+                return false;
+            }
+
+            var ent = food.Owner; //Get the entity of the ItemComponent.
+            _storage.Insert(ent);
+            UpdateUserInterface();
+            return true;
+
+        }
+
+        //This is required. It's 'cook'.
+        private void wzhzhzh()
+        {
+            _busy = true;
+            // Convert storage into Dictionary of ingredients
+            _solids.Clear();
+            foreach(var item in _storage.ContainedEntities)
```

Comment:
(Just as further evidence that you can enumerate `_storage.ContainedEntities` without a `.ToList()` call, you've done it here!)

Reply:
I was getting 'collection has been modified' runtime errors/crashes a lot so I tolisted for security way too much. Even in places I wasn't modifying the collection :thinking: